src/setupTests.ts

```diff
@@ -0,0 +1,3 @@
+import '@testing-library/jest-dom';
```

Comment:
Is it possible to add a basic github workflow with linting and possibly some low hanging fruit testing?

Reply:
About testing - this code i 99% about opening popups, sending messages between windows and opening external sites, which all aren't supported by testing frameworks. Only thing that don't have to be mocked is string concatenation (for url creation), in which tests doesn't give any benefits.

Reply:
At its most basic I thought we could add something like this:

When the fetch does not fail, it asserts that the response object is returned.
When the fetch does fail it asserts that obtainAuthCookie is called.
If obtainAuthCookie succeeds, it asserts that fetch is called a second time and returns the response object returned by the second fetch
If obtainAuthCookie fails, it asserts the caller is surfaced a reasonable error.
If the second fetch fails, it asserts the caller is surfaced a reasonable error.

I think you are right about testing obtainAuthCookie. I am not sure how that function could be tested.